k9mail/src/main/java/com/fsck/k9/view/MessageHeader.java

```diff
@@ -143,6 +149,31 @@ public void onClick(View view) {
         }
     }
 
+    @Override
+    public boolean onLongClick(View view) {
+        switch (view.getId()) {
+            case R.id.from:
+                onAddAddressesToClipboard(mMessage.getFrom());
+                break;
+            case R.id.to:
+                try {
```

Comment:
Having a `try`/`catch` block here makes the code harder to read and duplicates the error handling code. I suggest creating a new method that gets the recipients from a message and handles the exception.